plugin/option/option.go

```diff
@@ -28,6 +31,11 @@ func (p *Plugin) Name() string {
 // if they are requested
 func (p *Plugin) ServeDHCP(ctx context.Context, req, res *dhcpv4.DHCPv4) error {
 	if dhcpserver.Discover(req) || dhcpserver.Request(req) {
+		bootFile, err := p.GetBootFileOpt(ctx, req, res)
+		if err != nil {
+			return err
+		}
+		res.UpdateOption(*bootFile)
 		for code, value := range p.Options {
```

Comment:
Instead of always adding the `filename` option even if the user doesn't want to provide PXE booting we could make that "opt-in" by just specifying `option filename` in the configuration. 
If we want to make the boot-file names configurable we might also move this into a dedicated plugin.

What do you think?

Reply:
Yes, it's true, we don't need bootfile for each request.  I will change it 👍